src/Task.cc

```diff
@@ -3448,12 +3447,55 @@ static void create_mapping(Task *t, AutoRemoteSyscalls &remote, const KernelMapp
                real_file_name, device, inode, nullptr, &km);
 }
 
-static void apply_mm_map(AutoRemoteSyscalls& remote, const struct prctl_mm_map& map)
+static void apply_mm_map(AutoRemoteSyscalls& remote, const NativeArch::prctl_mm_map& map)
 {
-  AutoRestoreMem remote_mm_map(remote, (const uint8_t*)&map, sizeof(map));
-  int result = remote.syscall(syscall_number_for_prctl(remote.task()->arch()), PR_SET_MM,
-                            PR_SET_MM_MAP, remote_mm_map.get().as_int(),
-                            sizeof(map));
+#if defined(__x86_64__)
+  /* This fflush somehow makes the following prctl succeed
+     with a 64-bit rr under 64-bit kernel instead of giving a EPERM error.
+     What does that change to our process? Why is this not needed for x86? */
+  fflush(stderr);
+#endif
+
+  unsigned int expected_size = 0;
+  int result = prctl(PR_SET_MM, PR_SET_MM_MAP_SIZE, &expected_size);
```

Comment:
Unused arguments should be specified as zero, otherwise this prctl can fail. (may explain what the previous fflush accomplishes)

Reply:
That's it - giving zwo more zero arguments make the prtcl succeed. Thank you very much.